### PR TITLE
Implement battery status "charge state"

### PR DESCRIPTION
**Pull request by BazookaJoe1900**
Implement battery status "charge state" on mavlink battery status message

**Describe problem solved by this pull request**
the field charge_state is implemented on the uorb side of the code (named battery_state) but not being sent to the ground. 

**Describe your solution**
Pass the data from the uorb to mavlink side

**Test data / coverage**
simulator

**Additional context**
Also added the missing enum values of UNHEALTHY & CHARGING to the uorb enum, but there is no implementation for them for now

### CODE REVIEW

**Comment by mhkabir** [comment]
We can likely implement the charge state on the PX4 side by monitoring when the power module current measurement goes negative.

**Comment by BazookaJoe1900** [comment]
> We can likely implement the charge state on the PX4 side by monitoring when the power module current measurement goes negative.

you right, but that for other pr. I think

**Comment by BazookaJoe1900** [comment]
@mhkabir 
> Seems harmless,

Its harmless but let ground control to know the state of the battery, without needing to read the parameters. I am not sure how QGC decide to paint the battery icon to red, but this field should be the index for that.

> ... but I'm not sure about the semantics of the "charging" state being a "warning". Maybe better to rename the uorb message field to "state"?

That can be done, but it will require changes on other places that use the battery_status_s::BATTERY_WARNING_xxx. suce the land detector, the commander and so on.
I can do that. if you think its better...

**Comment by mhkabir** [comment]
Let's rename the field then.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -700,6 +700,31 @@ class MavlinkStreamBatteryStatus : public MavlinkStream
 				bat_msg.current_battery = (battery_status.connected) ? battery_status.current_filtered_a * 100 : -1;
 				bat_msg.battery_remaining = (battery_status.connected) ? ceilf(battery_status.remaining * 100.0f) : -1;
 
+				if (battery_status.warning == battery_status_s::BATTERY_WARNING_NONE) {
```
Comment (mhkabir) [inline src/modules/mavlink/mavlink_messages.cpp]: Use a switch-case?

**Comment by dagar** [comment]
Please carefully check for any logic that relies on the numeric value of the warning (now state) increasing with severity.

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -1931,7 +1931,7 @@ Commander::run()
 
 		// abort auto mode or geofence reaction if sticks are moved significantly
 		// but only if not in a low battery handling action
-		const bool not_in_low_battery_reaction = _battery_warning < battery_status_s::BATTERY_WARNING_CRITICAL;
+		const bool not_in_low_battery_reaction = _battery_state < battery_status_s::BATTERY_STATE_CRITICAL;
```
Comment (mhkabir) [inline src/modules/commander/Commander.cpp]: This.

File: src/modules/mavlink/mavlink_high_latency2.cpp [hunk 3]
```diff
@@ -283,7 +283,7 @@ bool MavlinkStreamHighLatency2::write_battery_status(mavlink_high_latency2_t *ms
 			updated = true;
 			_batteries[i].connected = battery.connected;
 
-			if (battery.warning > battery_status_s::BATTERY_WARNING_LOW) {
+			if (battery.state > battery_status_s::BATTERY_STATE_LOW) {
```
Comment (mhkabir) [inline src/modules/mavlink/mavlink_high_latency2.cpp]: This

**Comment by BazookaJoe1900** [comment]
@mhkabir with your and @dagar reviews I think its better to remove the UNHEALTHY and CHARGE states from the uorb, to revert the 'state' back to be 'warning'. so the number of the var is increasing when the battery health degrading.
there are more points other than the two that you ask me to look at that uses the warning as 'degrading'.

anyway, for now UNHEALTHY and CHARGE can't be set by the publisher. and when it will it can be put on other variable.

So basically on the last two commits I pushed, I reverted the warning to state.
If you think that is ok, I can squash all that mess to one commit...

Do you agree?

**Comment by BazookaJoe1900** [comment]
@dagar @mhkabir , any reason not to merge this?
